modules/lib/lib-project/src/main/resources/lib/xp/project.js

```diff
@@ -0,0 +1,173 @@
+ */
+exports.list = function () {
+    var bean = __.newBean('com.enonic.xp.lib.project.ListProjectsHandler');
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Adds permissions to an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {Object.<string, string[]>} params.permissions Project permissions to add. 1 to 5 properties where key is a role and value is an array of principals.
+ * @param {string} params.permissions.role - Role id (one of `owner`, `editor`, `author`, `contributor`, `viewer`)
+ * @param {Object[]} params.permissions.principals - Array of principals to add to this role
+
+ *
+ * @returns {Object} All current project permissions.
+ */
+exports.addPermissions = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.AddProjectPermissionsHandler');
+    bean.id = required(params, 'id');
+    bean.permissions = __.toScriptValue(params.permissions);
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Removes permissions from an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {Object.<string, string[]>} params.permissions Project permissions to delete. 1 to 5 properties where key is a role and value is an array of principals.
+ * @param {string} params.permissions.role - Role id (one of `owner`, `editor`, `author`, `contributor`, `viewer`)
+ * @param {Object[]} params.permissions.principals - Array of principals to delete from this role
+
+ *
+ * @returns {Object} All current project permissions.
+ */
+exports.removePermissions = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.RemoveProjectPermissionsHandler');
+    bean.id = required(params, 'id');
+    bean.permissions = __.toScriptValue(params.permissions);
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Toggles public/private read access for an existing Content Project.
+ * This will modify permissions on all of the content items inside the repository by adding or removing READ access for `system.everyone`.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {Object<string, boolean>} params.readAccess Read access.
+ * @param {boolean} params.readAccess.public Public read access (READ permissions for `system.everyone`).
+
+ *
+ * @returns {Object<string, boolean>} Current state of Read access.
+ */
+exports.modifyReadAccess = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.ModifyProjectReadAccessHandler');
+    bean.id = required(params, 'id');
+    bean.readAccess = __.toScriptValue(params.readAccess);
+    return __.toNativeObject(bean.execute());
+};
+
+function required(params, name) {
+    var value = params[name];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable Assigned to Object Injection Sink](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5511138)